Content.Shared/Materials/SharedMaterialReclaimerSystem.cs

```diff
@@ -110,6 +111,21 @@ public bool TryStartProcessItem(EntityUid uid, EntityUid item, MaterialReclaimer
             component.NextSound = Timing.CurTime + component.SoundCooldown;
         }
 
+        if (TryComp<ContainerManagerComponent>(item, out var contManager)){
+            foreach (var cont in contManager.Containers.Values)
+            {
+                var containedEntites = cont.ContainedEntities.ToArray();
```

Comment:
This shouldn't iterate over all containers, it has to be opt-in. Not all containers are there to store physical entities that can be dropped. E.g., "Actions" in the actions hotbar are also entities in containers.

Instead, this should behave more like entity destruction/breaking. I.e., raise an event, then let the storage & entity-storage systems handle that event.